src/Kiota.Builder/KiotaBuilder.cs

```diff
@@ -613,6 +614,8 @@ private void CreateOperationMethods(OpenApiUrlTreeNode currentNode, OperationTyp
             var returnType = voidType;
             if(operation.Responses.Any(x => x.Value.Content.ContainsKey(RequestBodyBinaryContentType)))
                 returnType = "binary";
+            else if (operation.Responses.Any(x => x.Value.Content.ContainsKey(RequestBodyPlainTextContentType)))
```

Comment:
I believe we can implement a better solution than that.
For all scalar types, assuming the schema is declared, we should be able to use (and parse at runtime) this scalar type (int, double, float, etc...). For complex types, we should default to string since their is no telling how the objects are serialized.

Reply:
In this branch of the execution, we are sure the `schema` is null (see line 609). 
I don't think we can handle that here. Yes?

Reply:
aaah I missed that part, you're correct, we don't have a schema, so let's treat it as string.
I thought the new path items causing issues were the raw counts (/$count) which do have a schema, but apparently it's something different.